packages/gatsby/src/utils/detect-port-in-use-and-prompt.ts

```diff
@@ -5,11 +5,11 @@ import prompts from "prompts"
 export const detectPortInUseAndPrompt = async (
   port: number
 ): Promise<number> => {
-  let foundPort = port
-  const detectedPort = await detectPort(port).catch((err: Error) =>
+  let foundPort = typeof port === `string` ? parseInt(port, 10) : port
```

Comment:
why did you have to add this code? TS states that the port has to be a number.

Reply:
See [last message](https://github.com/gatsbyjs/gatsby/pull/25863#issuecomment-660823247) I left in this PR. `gatsby-cli` types the port to string, it's unclear if it that's intentional(it's possible to use ports that aren't numbers, though I doubt anyone is doing that). TS isn't doing any enforcement or conversion despite the typing.

Reply:
ah oof. Seems like a mistake on the TS side then.. Wonder if we should fix this more globally. to make port always be a number.

Reply:
I pushed up a change to handle the case early. Glad we found this discrepancy!

Reply:
Should be fine, do you want that change added to this PR?